pkg/skaffold/deploy/status_check.go

```diff
@@ -74,23 +69,29 @@ func StatusCheck(ctx context.Context, defaultLabeller *DefaultLabeller, runCtx *
 	return getSkaffoldDeployStatus(syncMap)
 }
 
-func getDeployments(client kubernetes.Interface, ns string, l *DefaultLabeller) (map[string]int32, error) {
+func getDeployments(client kubernetes.Interface, ns string, l *DefaultLabeller, deadlineDuration time.Duration) (map[string]time.Duration, error) {
 	deps, err := client.AppsV1().Deployments(ns).List(metav1.ListOptions{
 		LabelSelector: l.K8sManagedByLabelKeyValueString(),
 	})
 	if err != nil {
 		return nil, errors.Wrap(err, "could not fetch deployments")
 	}
 
-	depMap := map[string]int32{}
+	depMap := map[string]time.Duration{}
 
 	for _, d := range deps.Items {
-		var deadline int32
-		if d.Spec.ProgressDeadlineSeconds == nil {
-			logrus.Debugf("no progressDeadlineSeconds config found for deployment %s. Setting deadline to %d seconds", d.Name, defaultStatusCheckDeadlineInSeconds)
-			deadline = defaultStatusCheckDeadlineInSeconds
-		} else {
-			deadline = *d.Spec.ProgressDeadlineSeconds
+		var deadline time.Duration
+		switch d.Spec.ProgressDeadlineSeconds {
+		case nil:
+			logrus.Debugf("no progressDeadlineSeconds config found for deployment %s. Setting deadline to %s", d.Name, deadlineDuration)
+			deadline = deadlineDuration
+		default:
+			if *d.Spec.ProgressDeadlineSeconds > int32(deadlineDuration.Seconds()) {
+				logrus.Debugf("progressDeadlineSeconds for deployment %s is %ds greater than %s on command line. Setting deadline to %s", d.Name, d.Spec.ProgressDeadlineSeconds, deadlineDuration, deadlineDuration)
```

Comment:
This debug log is a bit confusing to me, are you trying to log the difference between the two timings, or just say that you're defaulting to the command line?

Reply:
see my other comment, but I think we should just remove these logs